In [245]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats

In [246]:
df = pd.read_csv('C:/Users/HP/OneDrive/Desktop/dataSci/hw4/EmployeeAttrition.csv')

## 1.
#### H0: the average monthlyIncome of males is greater than females. 
#### Ha: the average monthlyIncome of males is not greater than females.

In [247]:
male = df[(df['Gender']=='Male')]['MonthlyIncome']
female = df[(df['Gender']=='Female')]['MonthlyIncome']

In [248]:
stats.ttest_ind(male, female)

Ttest_indResult(statistic=-1.2212617308870655, pvalue=0.22218303455087904)

### Conclusion:
P-Value = 0.222, P-Value > 0.05 => NOT reject Null hypothesis H0: the average monthlyIncome of males is greater than females and have no significant differences. it means that average monthlyIncome of males is not less than females

## 2.
#### H0: the WorkLifeBalance of Males is less than Females 
#### Ha: the WorkLifeBalance of Males is not less than Females

In [249]:
male = df[(df['Gender']=='Male')]['WorkLifeBalance']
female = df[(df['Gender']=='Female')]['WorkLifeBalance']

In [250]:
stats.ttest_ind(male, female)

Ttest_indResult(statistic=-0.10546787956064164, pvalue=0.9160189649366257)

### Conclusion:
P-Value = 0.916, P-Value > 0.05 => NOT reject Null hypothesis H0: the WorkLifeBalance of Males is less than Females and have no significant differences. Means that WorkLifeBalance of Males in the company is not greater than Females. 

## 3.
#### H0: the YearsAtCompany of Single is less than Married
#### Ha: the YearsAtCompany of Single is not less than Married

In [251]:
Single= df[(df['MaritalStatus']=='Single')]['YearsAtCompany']
Married = df[(df['MaritalStatus']=='Married')]['YearsAtCompany']

In [252]:
stats.ttest_ind(Single, Married)

Ttest_indResult(statistic=-2.5589985737464476, pvalue=0.010625532709863163)

### Conclusion:
P-Value = 0.01, P-Value < 0.05 => Reject Null hypothesis and accept Ha: the YearsAtCompany of Single is not less than Married and significant differences does exits. Means that the average YearsAtCompany of a Single person is greater or equal to the average YearsAtCompany of a Married person.

## 4.
#### H0: the EnvironmentalSatisfaction of Attrition=Yes is less than Attrition=No
#### Ha: the EnvironmentalSatisfaction of Attrition=Yes is not less than Attrition=No

In [253]:
yes = df[(df['Attrition']=='Yes')]['EnvironmentSatisfaction']
no = df[(df['Attrition']=='No')]['EnvironmentSatisfaction']

In [254]:
stats.ttest_ind(yes, no)

Ttest_indResult(statistic=-3.9818600026818496, pvalue=7.17233854936886e-05)

### Conclusion:
P-Value = 7.71e-5, P-Value < 0.05 => Reject Null hypothesis and accept Ha: the EnvironmentalSatisfaction of Attrition=Yes is not less than Attrition=No and and significant differences does exits.
it Means the people who wants to leave the company their EnvironmentalSatisfaction is greater or eqaul to the EnvironmentalSatisfaction of people who not wants to leave the company.

## 5.
#### H0: the average MonthlyIncome of Manager is greater than Laboratory Technician
#### Ha: the average MonthlyIncome of Manager is not greater than Laboratory Technician

In [255]:
manager = df[(df['JobRole']=='Manager')]['MonthlyIncome']
tech = df[(df['JobRole']=='Laboratory Technician')]['MonthlyIncome']

In [256]:
stats.ttest_ind(manager, tech)

Ttest_indResult(statistic=76.03317414634023, pvalue=1.9942336668736867e-223)

### Conclusion:
P-Value = 1.99e-223,which is very low than 0.05. P-Value < 0.05 => Reject Null hypothesis and accept Ha: the average MonthlyIncome of Manager is not greater than Laboratory Technician and significant differences does exit. so the average MonthlyIncome of manager is lesser or eqaul to Laboratory Technician but not greater.

## 6.
#### H0: there is no association between Department and Gender
#### Ha: there is an association between Department and Gender

In [257]:
dg = pd.crosstab(df['Department'],df['Gender'], margins=True)

In [258]:
value = np.array([dg.iloc[0][0:].values,dg.iloc[1][0:].values])

In [259]:
chi2, p, dof, ex = stats.chi2_contingency(value)

In [260]:
print(f"pvalue={p}")

pvalue=0.47927963769548776


## Conclusion.
p-value = 0.4792, p-value > 0.05 => we accept the NULL hypothesis and assume that the ‘Department’ and ‘Gender’ have no association and have no significant differences.
it means that "Department" and "Gender" are independent of each other.